README.md

```diff
@@ -353,6 +360,16 @@ The project includes several utility scripts for development and testing:
 # Monitor logs in real-time
 ./tools/tail-logs.sh status  # Show service status
 ./tools/tail-logs.sh all     # Tail all service logs
+
+ # Optional: Run E2E tests against a real backend (skipped by default)
+ # Choose exactly one backend using E2E_BACKEND to avoid conflicts.
+ # Milvus example:
+ # E2E_BACKEND=milvus E2E_MILVUS=1 MILVUS_URI=http://localhost:19530 \
+ # CUSTOM_EMBEDDING_URL=http://localhost:11434/v1 CUSTOM_EMBEDDING_MODEL=nomic-embed-text \
+ # CUSTOM_EMBEDDING_VECTORSIZE=768 pytest tests/e2e/test_mcp_milvus_e2e.py -m e2e -vv
+ # Weaviate example:
+ # E2E_BACKEND=weaviate E2E_WEAVIATE=1 WEAVIATE_API_KEY=... WEAVIATE_URL=... \
+ # pytest tests/e2e/test_mcp_weaviate_e2e.py -m e2e -vv
```

Comment:
given this is marked optional I see now reason to leave it in a comment, the user would just not run those lines, and based on the other content in this code block they wouldn't be copying and running the thing whole thing anyway

Reply:
I think the docs could still be improved, but we could iterate in a future PR.  Getting into CI asap is particularly useful as it will reduce the chance we'll break e2e function